Datastore/src/Connection/Grpc.php

```diff
@@ -242,7 +242,7 @@ public function runQuery(array $args)
                 $query['filter'] = $this->convertFilterProps($query['filter']);
             }
 
-            if (isset($query['limit']) && is_int($query['limit'])) {
+            if (isset($query['limit']) && !is_array($query['limit'])) {
```

Comment:
Vishwaraj explained to me that we need to check if the query is already an array, because otherwise we'd be wrapping it in an array again unnecessarily.